### ISSUE TITLE
@children decorator doesn't select "transcluded" elements

### ISSUE THREAD

**Issue by charlespockert** [aurelia/templating#218]
Example:

**ViewModel**
```
@children({ name: "items", selector: "div" })
export class MyViewModel {
}
```

**Template**
```
<template>
    <ul><li repeat.for="item of items">Item ${ $index }</li></ul>
    <content></content>
</template>
```

**Usage**
```
<my-view-model>
  <div>Hello</div>
  <div>World</div>
</my-view-model>
```

**Comment by charlespockert** [comment]
Actually, just running through it in debug, it looks more like no mutation records are detected since the elements are already in the DOM when the mutation observer is added.

Does this mean it just doesn't work at all unless the DOM mutates?

**Comment by EisenbergEffect** [comment]
No, there's an initial check that is done when it's bound which isn't dependent on mutation observers.

**Comment by charlespockert** [comment]
Ah yes I see that now, so because it only skims siblings, it never sees inside content selectors?

**Comment by charlespockert** [comment]
@EisenbergEffect After some poking around I can't see anything that would traverse any child elements at all, so as the update blog post says, this is limited to direct children only.

I've checked again and the above sample actually works fine - it only doesn't work when you have any sort of parent element wrapping the items you wish to select (even if they are part of the current view and you aren't crossing any view boundaries)

This poses a problem which is the opposite of the original problem:

**tab-set.html**
```
<template>
    <ul class="nav nav-tabs">
        <li role="presentation" click.delegate="$parent.selectTab(tab)" repeat.for="tab of tabs" class="${ $parent.selectedTab === tab ? 'active' : '' }">
            <a href="#">
                <span innerhtml.bind="tab.heading"></span>
            </a>
        </li>
    </ul>
   <div style="margin-top:8px"> <!-- Here is the issue -->
        <content></content>
    </div>
</template>
```

The above does not work because of the `div` wrapping the transcluded content, but I really want that DIV there, I don't expect to have to provide it every time I use my custom element (I'm explicitly applying a style here but I could just use a css class)

I suppose the point is, the change has prevented you from selecting into the DOM tree too deeply, with the drawback that you can only ever skim the surface of your templates DOM to look for elements...

**Comment by EisenbergEffect** [comment]
I think I've got a solution in mind.

**Comment by kiryaka** [comment]
Any updates? This bug making the creation of complex wrapper/decorator components fairly unpleasant. Workarounds looks like hacks most of the time. 
`<template><content></content></template>` works
`<template><div><content></content></div></template>` does not works

**Comment by EisenbergEffect** [comment]
Sorry this issue wasn't updated. I think it was mentioned in another issue. This bug will be solved, but probably not until we update the Shadow DOM to v1. The good news is that work on that starts next week. Hang in there!

**Comment by kiryaka** [comment]
Good to hear that it is in progress. Thank you for the quick reply. Will survive on the workarounds till there.

**Comment by EisenbergEffect** [comment]
This issue is fixed in our shadow dom v1 branch.
